connmgr.go

```diff
@@ -113,11 +134,17 @@ func NewConnManager(low, hi int, grace time.Duration) *BasicConnMgr {
 		}(),
 	}
 
+	decay, _ := NewDecayer(cfg.decayer, cm)
+	cm.decayer = decay
+
 	go cm.background()
 	return cm
 }
 
 func (cm *BasicConnMgr) Close() error {
+	if err := cm.decayer.Close(); err != nil {
+		return err
+	}
```

Comment:
I think we should still cancel the background goroutine, even if this fails.

Reply:
Makes sense.